features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/DesignConfigurations.jsx

```diff
@@ -195,24 +207,156 @@ export default function DesignConfigurations() {
     const [isUpdating, setIsUpdating] = useState(false);
     const [apiConfig, configDispatcher] = useReducer(configReducer, copyAPIConfig(api));
     const classes = useStyles();
+    const [descriptionType, setDescriptionType] = useState('');
+    const [overview, setOverview] = useState('');
+    const [overviewDocument, setOverviewDocument] = useState(null);
 
     const invalidTagsExist = apiConfig.tags.find((tag) => {
         return (/([~!@#;%^&*+=|\\<>"'/,])/.test(tag));
     });
+    const handleChange = (event) => {
+        const type = event.target.value;
+        if (type === CONSTS.DESCRIPTION_TYPES.DESCRIPTION) {
+            if (apiConfig.description === null) {
+                configDispatcher({ action: CONSTS.DESCRIPTION_TYPES.DESCRIPTION, value: overview });
+            }
+        } else if (type === CONSTS.DESCRIPTION_TYPES.OVERVIEW) {
+            if (overviewDocument === null) {
+                setOverview(apiConfig.description);
+            }
+        }
+        setDescriptionType(type);
+    };
+    const updateContent = (content) => {
+        if (descriptionType === CONSTS.DESCRIPTION_TYPES.DESCRIPTION) {
+            configDispatcher({ action: CONSTS.DESCRIPTION_TYPES.DESCRIPTION, value: content });
+        } else if (descriptionType === CONSTS.DESCRIPTION_TYPES.OVERVIEW) {
+            configDispatcher({ action: CONSTS.DESCRIPTION_TYPES.DESCRIPTION, value: null });
+            setOverview(content);
+        }
+    };
+    const loadContentForDoc = (documentId) => {
+        const { apiType } = api.apiType;
+        const restApi = apiType === API.CONSTS.APIProduct ? new APIProduct() : new API();
+        const docPromise = restApi.getInlineContentOfDocument(api.id, documentId);
+        docPromise
+            .then((doc) => {
+                let { text } = doc;
+                Object.keys(api).forEach((fieldName) => {
+                    const regex = new RegExp('___' + fieldName + '___', 'g');
+                    text = text.replace(regex, api[fieldName]);
+                });
+                setOverview(text);
+            })
+            .catch((error) => {
```

Comment:
This will mask the original exception, We should notify the user about the error and log it `console.error`

Reply:
Shall we simply reemovee this catch blok ? @hisanhunais

Reply:
Removed the catch block in https://github.com/wso2/carbon-apimgt/pull/9763/commits/bda6fa8be2dc46b2f647b495141183bac8ce3408